dallinger/config.py

```diff
@@ -120,6 +133,14 @@ def extend(self, mapping, cast_types=False, strict=False):
                         value=repr(value), key=key, expected_type=expected_type
                     )
                 )
+            for validator in self.validators.get(key, []):
+                try:
+                    validator(value)
+                except ValueError as e:
+                    # Annotate the exception with more info
+                    e.dallinger_config_key = key
```

Comment:
I would have done that if we didn't have to support Python 2, because then I could have just used `raise ... from e` and tracebacks would have been fine. My worry was that re-raising a new exception would mean we lose the traceback if it's encountered in custom code. I'm not sure if that's a realistic worry or not?

Reply:
Ah, I forgot about P2  (when can we drop support? ;-) ). I thought I was already using raise..from somewhere in mturk.py, but I don't see it. 

I see your point. If we moved to a more specific subclass of `ValueError` later, anyone who for some reason started `catch`ing `ValueError` now would still catch the new exception, so we could go with this for now (it's a pretty small nitpick) and change it later if we wanted to.

Reply:
To be honest, I don't feel it's worth making this change at this point. Definitely when we drop py2 though.